src/screens/connect-extension/connect-extension.tsx

```diff
@@ -29,42 +49,137 @@ export class ConnectExtensionScreenComponent extends React.Component<
 > {
     public static navigationOptions = navigationOptions;
     public qrCodeScanner: any;
-    public connectionInterval: any;
 
     constructor(props: IThemeProps<ReturnType<typeof stylesProvider>>) {
         super(props);
         this.state = {
-            isConnected: WalletConnectClient.isConnected()
+            isConnected: false,
+            isLoading: false,
+            os: undefined,
+            platform: undefined
         };
     }
 
     public componentDidMount() {
-        this.connectionInterval = setInterval(async () => {
+        this.getConnectionAsyncStorage();
+    }
+
+    private async getConnectionAsyncStorage() {
+        this.setState({ isLoading: true });
+
+        try {
+            const encryptionKey = await getBaseEncryptionKey();
+            const connection = await readEncrypted(CONN_EXTENSION, encryptionKey);
+
+            if (connection) {
+                const connectionParse = JSON.parse(connection);
+                this.setState({
+                    isConnected: true,
+                    os: connectionParse?.os,
+                    platform: connectionParse?.platform
+                });
+
+                await this.syncExtension(connectionParse);
+            }
+
+            this.setState({ isLoading: false });
+        } catch {
+            this.setState({ isLoading: false });
+        }
+    }
+
+    private async extractQrCodeData(value: string): Promise<IQRCode> {
+        const connection: IQRCode = {
+            connectionId: undefined,
+            encKey: undefined,
+            os: undefined,
+            platform: undefined
+        };
+
+        const res = qrCodeRegex.exec(value);
+        if (res) {
+            const extraData: any = getUrlParams(res[4]);
+
+            connection.connectionId = res[1];
+            connection.encKey = extraData?.encKey;
+            connection.os = extraData?.os;
+            connection.platform = extraData?.browser;
+        }
+
+        return connection;
+    }
+
+    private async syncExtension(connection: any) {
+        try {
+            const http = new HttpClient(CONFIG.extSyncUpdateStateUrl);
+            const res = await http.post('', {
+                connectionId: connection.connectionId,
+                data: await encrypt(
+                    JSON.stringify(extensionState(store.getState())),
+                    connection.encKey
+                ),
+                authToken: sha256(connection.encKey),
+                fcmToken: await Notifications.getToken()
+            });
+
+            return res;
+        } catch (err) {
+            throw err;
```

Comment:
we shouldn't throw errors in async functions, it would be better to return a `Promise.reject()`